include/sisl/sobject/sobject.hpp

```diff
@@ -0,0 +1,116 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <string>
+#include <functional>
+#include <map>
+#include <set>
+#include <memory>
+#include <shared_mutex>
+#include <vector>
+#include <nlohmann/json.hpp>
+
+#include <sisl/logging/logging.h>
+
```

Comment:
Some comments or example how type/name would look like would be better for understanding.